File: javascript/bob/bob.js
var Bob = function() {
  this.hey = function (phrase) {
    if (silence(phrase)) {
      return 'Fine. Be that way!'
    }
    else if (yelling(phrase)) {
      return 'Woah, chill out!'
    }
    else if (question(phrase)) {
      return 'Sure.'
    }
    else
      return 'Whatever.'
  }
  silence = function(phrase) {
    return phrase.match(/^\s*$/)
  }
  question = function(phrase) {
    return phrase.charAt(phrase.length-1) == '?'
  }
  yelling = function(phrase) {
    return phrase.toUpperCase() == phrase
  }
};
module.exports = Bob;

File: clojure/bob/bob.clj
(ns bob)
(use 'clojure.string)
(declare silent?)
(declare yelling?)
(declare question?)

(defn response-for [phrase]
  (cond
    (silent? phrase)   "Fine. Be that way!"
    (yelling? phrase)  "Woah, chill out!"
    (question? phrase) "Sure."
    :else "Whatever.")
)

(defn- silent? [phrase]
 (or (nil? phrase) (blank? phrase))
)

(defn- yelling? [phrase]
  (= (upper-case phrase) phrase)
)

(defn- question? [phrase]
  (= \? (last phrase))
)
